org.eclipse.jdt.ui.tests/ui/org/eclipse/jdt/ui/tests/buildpath/ClassPathElemenetsOrderTest.java

```diff
@@ -0,0 +1,166 @@
+				new Path("org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-17"), null);
+		cont1.createAttributeElement(CPListElement.MODULE, Boolean.TRUE.toString(), false);
+		elements.add(cont1);
+		CPListElement cont2= new CPListElement(javaProject, IClasspathEntry.CPE_CONTAINER,
+				new Path("org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/J2SE-1.3"), null);
+		cont2.createAttributeElement(CPListElement.MODULE, Boolean.TRUE.toString(), false);
+		elements.add(cont2);
+		CPListElement src1= new CPListElement(javaProject, IClasspathEntry.CPE_SOURCE, new Path("/hw2/src"), null);
+		elements.add(src1);
+		CPListElement src2= new CPListElement(javaProject, IClasspathEntry.CPE_SOURCE, new Path("/hw2/src2"), null);
+		elements.add(src2);
+		CPListElement lib1= new CPListElement(javaProject, IClasspathEntry.CPE_LIBRARY, new Path("/data/swt.jar"), null);
+		elements.add(lib1);
+		CPListElement lib2= new CPListElement(javaProject, IClasspathEntry.CPE_LIBRARY, new Path("/data/jface.jar"), null);
+		elements.add(lib2);
+		CheckedListDialogField<CPListElement> classPathList= new CheckedListDialogField<>(null, null, null);
+		classPathList.setElements(elements);
+		TestableLibrariesWorkBookPage page= new TestableLibrariesWorkBookPage(classPathList, null);
+		page.init(javaProject);
+		assertEquals("There should be 6 classpath elements before adding external jar", 6, classPathList.getElements().size());
+		assertEquals("java container should be at the top", cont1, classPathList.getElements().get(0));
+		assertEquals("src should be at position 2", src1, classPathList.getElements().get(2));
+		page.addExternalJar();
+		assertEquals("There should be 7 classpath elements after adding external jar", 7, classPathList.getElements().size());
+		assertEquals("java container should be at the top", cont1, classPathList.getElements().get(0));
+		assertEquals("src should be at position 2", src1, classPathList.getElements().get(2));
+		assertEquals("newly added jar must be at the end of the list", NEW_JAR_PATH, classPathList.getElements().get(6).getPath().toString());
+	}
+
+	public class TestableLibrariesWorkBookPage extends LibrariesWorkbookPage {
+
+		public TestableLibrariesWorkBookPage(CheckedListDialogField<CPListElement> classPathList, IWorkbenchPreferenceContainer pageContainer) {
+			super(classPathList, pageContainer);
+		}
+
+		public void addExternalJar() {
+			getLibrariesAdapter().customButtonPressed(null, IDX_ADDEXT);
+		}
+
+		@Override
+		protected CPListElement[] openExtJarFileDialog(CPListElement existing) {
+			IPath ele1Path= new Path(NEW_JAR_PATH);
+			CPListElement cpElement= new CPListElement(javaProject, IClasspathEntry.CPE_LIBRARY, ele1Path, null);
+			CPListElement[] list= new CPListElement[1];
+			list[0]= cpElement;
+			return list;
+		}
+
+		@Override
+		protected List<Object> getSelectedLibraryElements() {
```

Comment:
It is not clear why do we override the base method and change its behavior. If this is needed, please provide a comment in the code why. If it is not needed, please remove.